account/src/main/java/com/caldevsupplychain/account/controller/AccountController.java

```diff
@@ -0,0 +1,263 @@
+import io.swagger.annotations.Api;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.UsernamePasswordToken;
+import org.apache.shiro.subject.Subject;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.context.request.WebRequest;
+
+import javax.mail.MessagingException;
+import javax.servlet.Servlet;
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.Null;
+import javax.websocket.server.PathParam;
+import java.util.Locale;
+import java.util.Optional;
+
+@Slf4j
+@RestController
+@RequestMapping("/api/account/v1")
+@AllArgsConstructor
+public class AccountController {
+
+    private ModelMapper modelMapper;
+
+    private AccountService accountService;
+
+    private AccountWSBuilder accountWSBuilder;
+
+    /* validators */
+    private SignupValidator signupValidator;
+    private LoginValidator loginValidator;
+    private EditUserValidator editUserValidator;
+
+    private EmailService emailService;
+
+    // use validator to customize error code and message
+    @InitBinder("userSignupDTO")
+    public void initUserSignupFormBinder(WebDataBinder binder) {binder.addValidators(signupValidator);}
+    @InitBinder("userLoginDTO")
+    public void initUserLoginFormBinder(WebDataBinder binder) {binder.addValidators(loginValidator);}
+    @InitBinder("userEditDTO")
+    public void initUserEditFormBinder(WebDataBinder binder) {binder.addValidators(editUserValidator);}
+
+
+
+    /************************************************************************************************
+     |									Account API													|
+     ************************************************************************************************/
+
+
+    @GetMapping("/demo")
+    public ResponseEntity<?> demo (HttpServletRequest request) throws NullPointerException {
+
+        throw new NullPointerException("Throwing from demo");
+//        throw new IllegalAccessError("Illegal throw from dog");
+//        return null;
+    }
+
+
+    @PostMapping("/signup")
+    public ResponseEntity<?> signup (
+            @RequestParam(required = false, defaultValue = "USER") String role,
+            @Validated @RequestBody UserSignupDTO userSignupDTO,
+            HttpServletRequest request
+    ) throws MessagingException
```

Comment:
Don't throw exception on the controller. Catch any exception and map it to proper http response instead.

Reply:
Fixed